modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -4162,7 +4162,7 @@ else if (sqlType == Types.INTEGER) {
 				insertedValues, columnNames.get(index - 1), integer);
 		}
 		else if (sqlType == Types.VARCHAR) {
-			String string = String.valueOf(value);
+			String string = (String)value;
```

Comment:
@p-albuquerque can you test if this solution works when we have a text field and send a numeric value to the field?

Reply:
works as expected